### ISSUE TITLE
Error when both `n.chain` and `start.val` are given if `n.chain>1`

### ISSUE THREAD

**Issue by hdarjus** [ArdiaD/bayesGARCH#2]
Hi @ArdiaD,

thanks for the package. For improved MCMC diagnostics for my case study, I wanted to run multiple chains and also provide starting values. I got into an error that I was able to overcome only by reading the source code of the package.

MWE:

```r
model_control <- list(n.chain = 2,
                      start.val = matrix(c(0.01, 0.1, 0.7, 500,
                                           0.3, 3, 0.01, 500),
                                         2, 4, byrow = TRUE))
fit <- bayesGARCH(rep(0, 10), control = model_control)
```

The following error is thrown before sampling starts:

```
Error in dimnames(x) <- dn : 
  length of 'dimnames' [2] not equal to array extent
```

The problem is in this line: https://github.com/ArdiaD/bayesGARCH/blob/becc562a64cc816c9f28cdd33b472dde704fb5e0/R/sampler.R#L82

It is true that providing `n.chain` is redundant if `start.val` is given but I personally found the behavior counterintuitive (I think in JAGS and Stan you have to provide both) and the solution should be simple.
I hope you find this ticket useful.

Cheers,
Darjus

**Comment by ArdiaD** [comment]
Hi @hdarjus cannot you simple make a loop around the fit and collect the results?

**Comment by hdarjus** [comment]
Yes, that's one potential workaround. However, that doesn't solve the issue that I provided consistent `control` input (since `n.chain` is equal to `NROW(start.val)`) to `bayesGARCH` and received an error.

A package-wide solution could be to add something like

`else if (NROW(con$start.val) > 1) {if (NROW(con$start.val) != con$n.chain) stop("Arguments 'n.chain' and 'start.val' inconsistent; please double-check your input.")}`

to the above linked line before the `else`.

**Comment by ArdiaD** [comment]
Simply put n.chain = 1. I won't change the package for this design feature. But thx for your proposal

**Comment by ArdiaD** [comment]
Oh, sorry @hdarjus, I missunderstood your idea. Yes, will make the fix for the next release. Thx for pointing this to me !

**Comment by hdarjus** [comment]
I am glad it works out!
